jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,81 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
+        currently) if there are any finite values with non-finite uncertainties.
+
+        Note that if a the uncertainty array is FULLY non-finite and the data is
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
+    def _check_non_finite_uncertainty_mismatch(self, event={}):
+
+        if not hasattr(self, 'dataset') or self.dataset_selected == '':
+            # during initial init, this can trigger before the component is initialized
+            return
+
+        if not hasattr(self, '_get_1d_spectrum'):
+            # only model_fitting has _get_1d_spectrum(), but this method is here
+            # instead of there  because it may eventually be used by other plugins.
+            # if that happens, move _get_1d_spectrum() somewhere more general
+            m = "_get_1d_spectrum() must be available in plugin to use"
+            " NonFiniteUncertaintyMismatchMixin"
+            raise NotImplementedError(m)
+
+        spec = self._get_1d_spectrum()
+
+        if spec.uncertainty is None:
+            self.non_finite_uncertainty_mismatch = False
+            return self.non_finite_uncertainty_mismatch
+
+        if self.spectral_subset_selected == "Entire Spectrum":
+            flux = spec.flux
+            uncert = spec.uncertainty
+        else:
+            # get selected subset
+            spec = self._apply_subset_masks(self._get_1d_spectrum(), self.spectral_subset)
+            flux = spec.flux[~spec.mask]
+            uncert = spec.uncertainty[~spec.mask]
+
+        uncert = uncert.array
+
+        if not np.any(uncert):
+            self.non_finite_uncertainty_mismatch = False
+            return self.non_finite_uncertainty_mismatch
+
+        flux = flux.value
+
+        for i, val in enumerate(uncert):
+            if not np.isfinite(uncert[i]) and np.isfinite(flux[i]):
+                self.non_finite_uncertainty_mismatch = True
+                # we only need to catch this once then can stop looking
+                return self.non_finite_uncertainty_mismatch
+
+        # if we didn't find a mismatch, set to False
+        self.non_finite_uncertainty_mismatch = False
+        return self.non_finite_uncertainty_mismatch
```

Comment:
actually, maybe we can avoid the for-loop here entirely!

```suggestion
        self.non_finite_uncertainty_mismatch = np.any(np.logical_and(~np.isfinite(uncert) and np.isfinite(flux)))
        return self.non_finite_uncertainty_mismatch
```

Reply:
it looks like np.any(np.logical_and(a, b)) works

Reply:
oh i learned something new and annoying, the return value of np.any(np.logical_and(~np.isfinite(uncert) and np.isfinite(flux))) is a numpy bool, which traitlets didn't like, (and doesn't equate to True with 'is True', but does with '== True')

Reply:
interesting!  Casting to bool it is then!